src/layout.ts

```diff
@@ -44,24 +44,42 @@ export type LinkObject = {
   url: string;
 };
 
-export function layoutPage(content: Paragraph[], box: Box, fonts: Font[]): Frame {
+export function layoutPages(paragraphs: Paragraph[], box: Box, fonts: Font[]): Frame[] {
+  const pages = [];
+  let remainingParagraphs = paragraphs;
+  while (remainingParagraphs?.length) {
+    const { frame, remainder } = layoutPage(remainingParagraphs, box, fonts);
+    remainingParagraphs = remainder;
+    pages.push(frame);
+  }
+  return pages;
+}
+
+export function layoutPage(content: Paragraph[], box: Box, fonts: Font[]) {
   const { x, y, width, height } = box;
   const children = [];
   const pos = { x: 0, y: 0 };
   let lastMargin = 0;
   let remainingHeight = height;
-  content.forEach((paragraph) => {
+  let remainder;
+  for (const [idx, paragraph] of content.entries()) {
     const margin = paragraph.margin ?? ZERO_EDGES;
     const topMargin = Math.max(lastMargin, margin.top);
     lastMargin = margin.bottom;
     const nextPos = { x: pos.x + margin.left, y: pos.y + topMargin };
     const maxSize = { width: width - margin.left - margin.right, height: remainingHeight };
     const frame = layoutParagraph(paragraph, { ...nextPos, ...maxSize }, fonts);
+    // If the first paragraph does not fit on the page, render it anyway.
+    // It wouldn't fit on the next page as well, ending in an endless loop.
+    if (remainingHeight < topMargin + frame.height && idx) {
+      remainder = content.slice(idx);
```

Comment:
There seems to be no tests for `layoutPage` returning the remainder or ignoring the remaining height. Is this on purpose?

Reply:
Right, I've missed that. Added a test. Thanks!